maven/src/main/java/org/commonjava/maven/galley/maven/model/view/MavenPomView.java

```diff
@@ -1047,8 +1047,8 @@ public String resolveExpressions( final String value, final String... activeProf
             if ( ssi == null )
             {
                 ssi = new StringSearchInterpolator();
-                ssi.addValueSource( new MavenPomViewVS( this, activeProfileIds ) );
             }
+            ssi.addValueSource( new MavenPomViewVS( this, activeProfileIds ) );
```

Comment:
I guess the first time we try to retrieve a value from the pom and it resolves expressions, this will be set, and the list of active profiles will be entered. This will affect future expression resolving calls, regardless of whether the profileId lists for those calls are different.

What this means is that we're keeping state at the MavenPomView level that doesn't belong there.

Consider what will happen if we add the list of profileIds when resolving an expression in a RepositoryView that is contained in a profile. In this case, we will start looking in the profile for expression values, which is what we want for that case. BUT, what happens if the next call is to retrieve the version of a dependency that's *not* contained in a profile, and that version has an expression in it? In this case, the expression resolver will have the opposite problem...it might resolve a property value from the profile when we shouldn't be looking at that profile.

The solution here is to get rid of the StringSearchInterpolator field, and instead construct one in this method. Just doing this will fail though, because if you have two properties like this:
```
<field-one>${field-two}</field-one>
<field-two>${field-one}</field-two>
```

and some dependency that uses the version `${field-one}`, then the method call will recurse infinitely. StringSearchInterpolator is smart enough to prevent this, but only if we do one of two things:

1. use the same SSI
2. maintain a single instance of RecursionInterceptor, and pass that into all calls

The recursion can happen via MavenPomViewVS, which calls resolveXPathExpression(..) which calls resolveExpressions(..) (this method). 

I think we need to make this method setup a new RecursionInterceptor (really `PrefixAwareRecursionInterceptor`), then immediately call a new copy of this method that declares the RecursionInterceptor as a parameter. This will pass the interceptor into the MavenPomViewVS ctor. Then, we add this interceptor as a parameter in resolveXPathExpression() as well, and call the new form of resolveExpressions from that, passing the interceptor around the whole time.

This will allow us to declare a local instance of StringSearchInterpolator in resolveExpressions (the new one variant), and call interpolate with the interceptor as the second parameter.